controllers/helpers.go

```diff
@@ -495,29 +504,48 @@ func DeleteCategories(ctx context.Context, client *nutanixClientV3.Client, categ
 				return errorMsg
 			}
 		}
-		// check if there are remaining category values
-		categoryKeyValues, err := client.V3.ListCategoryValues(ctx, key, &nutanixClientV3.CategoryListMetadata{})
-		if err != nil {
-			errorMsg := fmt.Errorf("failed to get values of category with key %s: %v", key, err)
-			klog.Error(errorMsg)
-			return errorMsg
-		}
-		if len(categoryKeyValues.Entities) > 0 {
-			errorMsg := fmt.Errorf("cannot remove category with key %s because it still has category values assigned", key)
-			klog.Error(errorMsg)
-			return errorMsg
-		}
-		klog.Infof("No values assigned to category. Removing category with key %s", key)
-		err = client.V3.DeleteCategoryKey(ctx, key)
-		if err != nil {
-			errorMsg := fmt.Errorf("failed to delete category with key %s: %v", key, err)
-			klog.Error(errorMsg)
-			return errorMsg
+
+		if !ignoreKeyDeletion {
+			// check if there are remaining category values
+			categoryKeyValues, err := client.V3.ListCategoryValues(ctx, key, &nutanixClientV3.CategoryListMetadata{})
+			if err != nil {
+				errorMsg := fmt.Errorf("failed to get values of category with key %s: %v", key, err)
+				klog.Error(errorMsg)
+				return errorMsg
+			}
+			if len(categoryKeyValues.Entities) > 0 {
+				errorMsg := fmt.Errorf("cannot remove category with key %s because it still has category values assigned", key)
+				klog.Error(errorMsg)
+				return errorMsg
+			}
+			klog.Infof("No values assigned to category. Removing category with key %s", key)
+			err = client.V3.DeleteCategoryKey(ctx, key)
+			if err != nil {
+				errorMsg := fmt.Errorf("failed to delete category with key %s: %v", key, err)
+				klog.Error(errorMsg)
+				return errorMsg
+			}
 		}
 	}
 	return nil
 }
 
+// DeleteCategories deletes the given list of categories
+func DeleteCategories(ctx context.Context, client *nutanixClientV3.Client, categoryIdentifiers, obsoleteCategoryIdentifiers []*infrav1.NutanixCategoryIdentifier) error {
```

Comment:
Can't we just make this a generic function that requires one list of categoryIdentifiers?